types/p-defer/p-defer-tests.ts

```diff
@@ -1,13 +1,16 @@
 import pDefer = require('p-defer');
 
-function delay(deferred: pDefer.DeferredPromise<string>,  ms: number) {
+function delay(deferred: pDefer.DeferredPromise<string>, ms: number) {
     setTimeout(deferred.resolve, ms, '🦄');
     return deferred.promise;
 }
 
 let s: string;
-async function f() { s = await delay(pDefer<string>(), 100); }
+async function f() {
+    s = await delay(pDefer<string>(), 100);
+}
 
 async function u() {
-	const u: Promise<any> = pDefer().resolve();
+    const u: void = pDefer().resolve();
```

Comment:
Rather than doing this to enforce that the result of `resolve()` is void and having to disable the tslint rules, you can do the following:
```
// $ExpectType void
pDefer().resolve();
```
This will check the return types of resolve and reject as what you have currently.
See [here](https://github.com/Microsoft/dtslint#write-tests) for more info.